hw3/ThronesAPI/src/CharacterPage.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+import Searchbar from "./Searchbar";
+import { Container, Card } from "react-bootstrap";
+import axios from "axios";
+
+const url = "https://thronesapi.com/api/v2/Characters";
+
+const CharacterPage = () => {
+  const [characters, setCharacters] = useState([]);
+  const [searchResults, setSearchResults] = useState([]);
+
+  useEffect(() => {
+    const fetchCharacters = async () => {
+      try {
+        const response = await axios.get(url);
+        const apiData = response.data;
+        setCharacters(apiData);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    fetchCharacters();
+  }, []);
+
+  return (
+    <div>
+      <Searchbar characters={characters} setSearchResults={setSearchResults} />
+
+      {searchResults.map((character, index) => (
```

Comment:
Up to current React standards and clear! Nice work using useEffect and useState along with map() to search through these characters. I think this is the best way to use React as far as I know. You can't use for loops so map is the only way. Funny how React uses so many functional programming practices.